tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -558,3 +584,114 @@ def prepare_first_guess(
             index += atomic_number + 1
         first_guess[-1] = electron_density
         return first_guess
+
+    def population_objective_function(
+        self,
+        populations,
+        atomic_numbers,
+        phi,
+        solution_vector,
+        rate_matrix_index,
+        total_photo_ion_coefficients,
+        total_rad_recomb_coefficients,
+        total_coll_ion_coefficients,
+        total_coll_recomb_coefficients,
+    ):
+        """Main set of equations for the NLTE ionization solver. A*x - B = 0, where x is the populations,
+        A is the matrix of rates and B is the solution vector.
```

Comment:
```suggestion
        """Main set of equations for the NLTE ionization solver. 
        
        To solve the statistical equilibrium equations, we need to find the root 
        of the objective function A*x - B, where x are the populations, 
        A is the matrix of rates, and B is the solution vector.
```
I am not a hundred percent happy with this but it could be a starting point for you to work on it.